README.md

```diff
@@ -1,2 +1,127 @@
 #Idea Town
-The addon where ideas come to idea
\ No newline at end of file
+The addon where ideas come to idea
+
+## installation
+
+`npm install`
+
+## configuration
+
+check the `CONFIG` property in `package.json`
+
+## running
+
+* `npm start`
+* open http://localhost:1337/stub-content/index.html
+
+## packaging
+
+`npm run package`
+
+## Events
+
+Accepted:
+* `update-check`
+* `update-approve`
+* `loaded`
+
+Emitted:
+* `addon-available`
+* `addon-updates`
+* `addon-install:install-started`
+* `addon-install:install-new`
+* `addon-install:install-cancelled`
+* `addon-install:install-ended`
+* `addon-install:install-failed`
+* `addon-install:download-started`
+* `addon-install:download-progress`
+* `addon-install:download-ended`
+* `addon-install:download-cancelled`
+* `addon-install:download-failed`
+* `addon-self:installed`
+* `addon-self:enabled`
+* `addon-self:upgraded`
+
+Any emitted events prefixed with `addon-install:` will have an associated object
+which will be structured as such:
+
+``` json
+{
+  "name": "Fox Splitter",
+  "error": 0,
+  "state": 6,
+  "version": "2.1.2012122901.1-signed",
+  "progress": 517308,
+  "maxProgress": 517308
+}
+
+```
+The event `addon-install:install-ended` will include some extra properties:
+
+``` json
+{
+  "name": "Fox Splitter",
+  "error": 0,
+  "state": 6,
+  "version": "2.1.2012122901.1-signed",
+  "progress": 517308,
+  "maxProgress": 517308,
+  "description": "Splits browser window as you like.",
+  "homepageURL": "http://piro.sakura.ne.jp/xul/foxsplitter/index.html.en",
+  "iconURL": "file:///tmp/074a4b62-239e-49bb-b75a-4935c349855c/extensions/foxsplitter@piro.sakura.ne.jp/icon.png",
+  "size": 511221,
+  "signedState": 2,
+  "permissions": 13
+}
+```
+
+#### Talking to the addon
+
+You will need to setup the following function (or an equivalent) to send messages to the addon.
+
+``` javascript
+function sendToAddon (data) {
+  document.documentElement.dispatchEvent(new CustomEvent(
+    'from-web-to-addon', { bubbles: true, detail: data }
+  ));
+}
```

Comment:
> hmm, I feel like this would maybe put more pressure on us to use more unique event names in order to prevent against collisions.

I don't follow. It seems like it's basically the same problem in both cases: either we vary the name of the event, or the `type` of the event.